src/BSON/Iterator.c

```diff
@@ -263,10 +298,92 @@ static HashTable* php_phongo_iterator_get_properties(phongo_compat_object_handle
 	return php_phongo_iterator_get_properties_hash(object, false);
 }
 
+/* Iterator handlers */
+static void php_phongo_iterator_it_dtor(zend_object_iterator* iter)
```

Comment:
This entire logic was shamelessly lifted from [spl_array.c](https://github.com/php/php-src/blob/633e7455e55cb734acea13a14d73b8e976ad2158/ext/spl/spl_array.c#L958..L1048)

Reply:
Does "this entire logic" refer to _all_ of the iterator functions in this file (not just `dtor` here)?

Reply:
Yes - all of the `_it_*` methods were heavily inspired by PHP's standard implementation.